src/main/java/cr0s/warpdrive/AndroidCheck.java

```diff
@@ -0,0 +1,10 @@
+package cr0s.warpdrive;
+
+import matteroverdrive.entity.player.MOPlayerCapabilityProvider;
+import net.minecraft.entity.player.EntityPlayer;
+
+public class AndroidCheckHelper {
+	public static boolean isAndroid(EntityPlayer player) {
+		return MOPlayerCapabilityProvider.GetAndroidCapability(player) != null && MOPlayerCapabilityProvider.GetAndroidCapability(player).isAndroid();
```

Comment:
Capability is recovered twice, directly affecting performance. Consider storing the result in a local.